packages/autocomplete/Autocomplete.svelte

```diff
@@ -0,0 +1,309 @@
+<script>
+  import { exclude } from "@smui/common/exclude.js";
+  import { prefixFilter } from "@smui/common/prefixFilter.js";
+  import Textfield from "@smui/textfield";
+  import Menu from "@smui/menu";
+  import List, { Item, Text } from "@smui/list";
+  import { Anchor } from "@smui/menu-surface";
+  import { createEventDispatcher } from "svelte";
+
+  export let options = [];
+  export let value = undefined;
+  export let getOptionDisabled = undefined;
+  export let getOptionLabel = (option) => option || "";
+  export let blurOnSelect = true;
+  export let clearOnBlur = undefined;
+  export let clearOnEscape = true;
+  export let toggle = false;
+  export let dirty = undefined;
+  export let invalid = undefined;
+  export let freeSolo = false;
+
+  export let menu$class = "";
+  export let menu$anchor = false;
+  export let menu$anchorCorner = "BOTTOM_LEFT";
+
+  const dispatch = createEventDispatcher();
+
+  let items = [];
+  let loading = false;
+  let error = false;
+  let menu;
+  let menuIsOpen;
+  let anchorElement;
+  let formattedValue = "";
+  let matches = [];
+  let focusedIndex = -1;
+  let activeItems;
+  let focusedItem;
+  let activeOptions = [];
+  let shouldClearOnBlur = true;
+
+  $: textfieldProps = exclude($$props, [
+    "menu$",
+    "list$",
+    "value",
+    "options",
+    "getOptionDisabled",
+    "getOptionLabel",
+    "blurOnSelect",
+    "clearOnBlur",
+    "clearOnEscape",
+    "toggle",
+    "dirty",
+    "invalid",
+    "freeSolo",
+    "search",
+    "getOptionSelected",
+  ]);
+
+  $: (async () => {
+    if (typeof options == "function") {
+      items = await options();
+    } else {
+      items = options || [];
+    }
+  })();
+
+  export let search = (allOptions, input) =>
+    allOptions.filter((item) =>
+      getOptionLabel(item).toLowerCase().includes(input.toLowerCase())
```

Comment:
Ideally, we should expose in the API a filter function so that the client can optionally hook a filtering method of its choice. This would be the fallback/default filtering method. What do you think?

Reply:
I'm not sure I fully understand what you mean. Isn't it like that already? You can pass your custom filtering function through the `search` prop and if not this is the default one.

Reply:
Yes, your totally right, missed the `export` there 😄